src/preact/base-element.js

```diff
@@ -337,6 +370,11 @@ PreactBaseElement['Component'] = function () {
   devAssert(false, 'Must provide Component');
 };
 
+/**
+ * @protected {!Array<!ContextProp>}
+ */
+PreactBaseElement['useContexts'] = [];
```

Comment:
Nit:

```suggestion
PreactBaseElement['useContexts'] = getMode().localDev ? Object.freeze([]) : [];
```

Reply:
Ok. Applied this. But overall, I'm a bit less concerned about freezing these declarative structures.